syntax = "proto3";

import "ai/lzy/v1/zygote.proto";
import "google/protobuf/timestamp.proto";

package ai.lzy.v1;

option java_outer_classname = "LWB";
option java_package = "ai.lzy.v1";

message Whiteboard {
  string id = 1;
  string name = 2;
  repeated string tags = 3;
  repeated WhiteboardField fields = 4;
  Storage storage = 5;
  string namespace = 6;
  Status status = 7;
  google.protobuf.Timestamp createdAt = 8;

  enum Status {
    WHITEBOARD_STATUS_UNSPECIFIED = 0;
    CREATED = 1;
    FINALIZED = 2;
  }
}

message WhiteboardField {
  string fieldName = 1;

  oneof state {
    CreatedField createdState = 2;
    LinkedField linkedState = 3;
  }

  message CreatedField {
  }

  message LinkedField {
    DataScheme scheme = 1;
    string storageUri = 2;
  }
}

message Storage {
  string name = 1;
  string description = 2;
}

message TimeBounds {
  google.protobuf.Timestamp from = 1;
  google.protobuf.Timestamp to = 2;
}
